nix/default.nix

```diff
@@ -1,194 +1,76 @@
-          bytesmith = let
-            src = builtins.fetchGit {
-              url = "https://github.com/teto/bytesmith";
-              ref = "ghc92";
-              rev = "dfdf6922e118932ba7d671e05a5b65998349cee8";
-            };
-          in (hpPrev.callCabal2nix "bytesmith" src { });
-
-          weeder = let
-            src = builtins.fetchGit {
-              url = "https://github.com/ocharles/weeder";
-              ref = "master";
-              rev = "c58ed2a8c66dcf0b469f8343efb6b6f61c7c40f3";
-            };
-          in (hpPrev.callCabal2nix "weeder" src { });
-
-          # Master version for ghc-9.2
-          bytebuild = let
-            src = builtins.fetchGit {
-              url = "https://github.com/byteverse/bytebuild";
-              ref = "master";
-              rev = "dde5a9b07d03f5a9c33eba6d63ca0140ab6f406e";
-            };
-          in (pkgs.haskell.lib.dontCheck
-            (hpPrev.callCabal2nix "bytebuild" src { }));
-
-          # ghc-9.2 fix proposed: https://github.com/andrewthad/scientific-notation/pull/4
-          scientific-notation = let
-            src = builtins.fetchGit {
-              url = "https://github.com/andrewthad/scientific-notation";
-              ref = "master";
-              rev = "0076fe9ed83e70ce068962fbd1b49d475af7a7f2";
-            };
-          in (pkgs.haskell.lib.dontCheck
-            (hpPrev.callCabal2nix "scientific-notation" src { }));
-
-          # nixpkgs somehow doesn't fetch, use direct src
-          servant = mk-servant-lib hpPrev "";
-          servant-foreign = mk-servant-lib hpPrev "-foreign";
-          servant-server = mk-servant-lib hpPrev "-server";
-
-          # nixpkgs version are broken, use direct src
-          morpheus-graphql-tests = mk-morpheus-lib hpPrev "tests";
-          morpheus-graphql-core = mk-morpheus-lib hpPrev "core";
-          morpheus-graphql-code-gen = mk-morpheus-lib hpPrev "code-gen";
-          morpheus-graphql-client = mk-morpheus-lib hpPrev "client";
-
-          gerrit = let
-            src = builtins.fetchGit {
-              url =
-                "https://softwarefactory-project.io/r/software-factory/gerrit-haskell";
-              ref = "refs/changes/41/24541/1";
-              rev = "c0c337bccb35e7d94e6125ab034cfcc9efe68476";
-            };
-          in pkgs.haskell.lib.dontCheck (hpPrev.callCabal2nix "gerrit" src { });
-
-          # Relude needs head for ghc-9.2
-          relude = (pkgs.haskell.lib.overrideCabal hpPrev.relude {
-            broken = false;
-            src = builtins.fetchGit {
-              url = "https://github.com/kowainik/relude";
-              ref = "main";
-              rev = "fcb32257a37e34a48f70ddd55424394de4bb025c";
-            };
-          });
-
-          monocle = (hpPrev.callCabal2nix "monocle" self { }).overrideAttrs
-            (_: { MONOCLE_COMMIT = builtins.getEnv "MONOCLE_COMMIT"; });
-        };
-      };
+  # Add monocle and patch broken dependency to the haskell package set
```

Comment:
Note that even though we are using hspkgs, we can still override packages when needed, for example to pin bloodhound and morpheus to an older version. Though I'd be happy to move the overrides to hspkgs upstream when it makes sense.